package/MDAnalysis/guesser/base.py

```diff
@@ -0,0 +1,199 @@
+        the universe's guess_topologyAttributes API works.
+        See :ref:`guess_TopologyAttributes <guess_TopologyAttributes>`).
+    **kwargs: to pass additional data to the guesser that can be used with
+              different methos.
+
+    .. versionadded:: 2.7.0
+
+    """
+    context = 'base'
+    _guesser_methods: Dict = {}
+
+    def __init__(self, universe=None, **kwargs):
+        self._universe = universe
+        self._kwargs = kwargs
+
+    def update_kwargs(self, **kwargs):
+        self._kwargs.update(kwargs)
+
+    def copy(self):
+        """Return a copy of this Guesser"""
+        kwargs = copy.deepcopy(self._kwargs)
+        new = self.__class__(universe=None, **kwargs)
+        return new
+    
+    def are_guessable(self, attrs_to_guess):
+        """check if the passed atrributes can be guessed by the guesser class
+
+        Parameters
+        ----------
+        guess: list
+            Atrributes to be guessed then added to the universe
+
+        Returns
+        -------
+        Boolean value
+        """
+        if attrs_to_guess:
+            for a in attrs_to_guess:
+                if a.lower() not in self._guesser_methods:
+                    return False
+            return True
+
+    def guess_attr(self, attr_to_guess, force_guess=False):
+        """map the attribute to be guessed with the apporpiate guessing method
+
+        Parameters
+        ----------
+        attr_to_guess: list
+            an atrribute to be guessed then to be added to the universe
+        force_guess: boolean
+            To indicate wether to only partialy guess the empty values of the
+            attribute or to overwrite all existing values by guessed one
+
+        Returns
+        -------
+        list of guessed values
+
+        Raises
+        ------
+        ValueError
+            Failed to find or guess parent attribute
+
+        """
+
+        # check if the topology already has the attribute to partially guess it
+        if hasattr(self._universe.atoms, attr_to_guess) and not force_guess:
+            attr_values = np.array(getattr(self._universe.atoms, attr_to_guess, None))
+
+            empty_values = []
+            top_attr = _TOPOLOGY_ATTRS[attr_to_guess]
+
+            for a in attr_values:
+                value_missing = top_attr.is_value_missing(a)
+                empty_values.append(value_missing)
+
+            if True in empty_values:
+                # pass to the guesser_method indecies of attributes that have
+                # empty values to be guessed
+                attr_values[empty_values] = self._guesser_methods[attr_to_guess](
+                    partial_guess=empty_values)
+                return attr_values
+
+            else:
+                logger.info(
+                    f'There is no empty {attr_to_guess} values. Guesser did '
+                    'not guess any new values for {attr_to_guess} attribute')
```

Comment:
```suggestion
                    f'not guess any new values for {attr_to_guess} attribute')
```

Each separate f-string needs to be defined as such :-)